File: java/sprykerFileCreator/model/finder/ClassFinder.java
package sprykerFileCreator.model.finder;

import com.intellij.openapi.project.Project;
import com.jetbrains.php.PhpIndex;
import com.jetbrains.php.lang.psi.elements.PhpClass;
import org.jetbrains.annotations.NotNull;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassInterface;
import pav.sprykerFileCreator.model.renderer.dto.PhpClassItem;

import java.util.ArrayList;
import java.util.Collection;

public class ClassFinder implements ClassFinderInterface {
    private Project project;

    public ClassFinder(@NotNull Project project) {
        this.project = project;
    }

    @Override
    public Boolean doesClassExist(String fullQualifiedName) {
        Collection<PhpClass> phpClassCollection = this.getPhpClassCollection(fullQualifiedName);

        return phpClassCollection.size() > 0;
    }

    @Override
    public PhpClassInterface findClass(String fullQualifiedName) {
        Collection<PhpClass> phpClassCollection = this.getPhpClassCollection(fullQualifiedName);

        if (phpClassCollection.size() > 0) {
            PhpClass phpClass = phpClassCollection.iterator().next();
            String parentName = phpClass.getName();
            String namespace = phpClass.getNamespaceName();

            return new PhpClassItem(namespace, parentName);
        }

        return null;
    }

    @NotNull
    private Collection<PhpClass> getPhpClassCollection(String fullQualifiedName) {
        try {
            PhpIndex phpIndex = PhpIndex.getInstance(this.project);
            return phpIndex.getAnyByFQN(fullQualifiedName);
        } catch (IllegalStateException exception) {
            return new ArrayList<>();
        }
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/finder/ClassFinder.kt
package com.github.tobi812.sprykerplugin.models.finder

import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassInterface
import com.github.tobi812.sprykerplugin.models.renderer.dto.PhpClassItem
import com.intellij.openapi.project.Project
import com.jetbrains.php.PhpIndex
import com.jetbrains.php.lang.psi.elements.PhpClass
import java.util.ArrayList

class ClassFinder(private val project: Project) : ClassFinderInterface {
    override fun doesClassExist(fullQualifiedName: String): Boolean {
        val phpClassCollection = findPhpClassCollection(fullQualifiedName)
        return phpClassCollection.isNotEmpty()
    }

    override fun findClass(fullQualifiedCoreName: String): PhpClassInterface? {
        val phpClassCollection = this.findPhpClassCollection(fullQualifiedCoreName)
        if (phpClassCollection.isNotEmpty()) {
            val phpClass = phpClassCollection.iterator().next() ?: return null

            val parentName = phpClass.name
            val namespace = phpClass.namespaceName

            return PhpClassItem(namespace, parentName)
        }

        return null
    }

    override fun findPhpClassCollection(fullQualifiedName: String): Collection<PhpClass?> {
        return try {
            val phpIndex = PhpIndex.getInstance(project)
            phpIndex.getAnyByFQN(fullQualifiedName)
        } catch (exception: IllegalStateException) {
            ArrayList()
        }
    }
}
